streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordQueue.java

```diff
@@ -38,7 +38,7 @@
     private final TimestampExtractor timestampExtractor;
     private final TopicPartition partition;
     private final ArrayDeque<StampedRecord> fifoQueue;
-    private final TimestampTracker<ConsumerRecord<Object, Object>> timeTracker;
+    private long prevTimestamp = TimestampTracker.NOT_KNOWN;
```

Comment:
I think, we can remove interface `TimestampTracer` and class `MinTimestampTracker`